component/componenttest/nop_host.go

```diff
@@ -32,10 +32,10 @@ func (nh *nopHost) GetFactory(_ component.Kind, _ component.Type) component.Fact
 	return nil
 }
 
-func (nh *nopHost) GetExtensions() map[component.ID]component.Extension {
+func (nh *nopHost) GetExtensions() map[component.ID]component.Component {
```

Comment:
This would impact anyone who's implemented their own Host correct? Any thoughts on the correct deprecation path for this?

Reply:
Can we convert  `component.[Extension|Processor|...]` into deprecated aliases of `component.Component` for some time?

Reply:
This wouldn't change the breakage that would happen when the update to the `Host` interface happens... short of adding a new `WithComponent` interface and a new `Start` func, i'm not sure how to move forward without breaking this. I believe @tigrannajaryan hit something similar here: https://github.com/open-telemetry/opentelemetry-collector/pull/6560#discussion_r1025300926

Reply:
I actually missed that only `Exporter` is removed in this PR. I assumed the `Extension` going away as well

Reply:
> This wouldn't change the breakage that would happen when the update to the Host interface happens...

It's not a breaking change when parts of API are replaced with identical (aliased) types. We did that a lot in pdata.

Reply:
> It's not a breaking change when parts of API are replaced with identical (aliased) types. We did that a lot in pdata.

It is, because now we will have `type Extension = Component` then I want to revert that to 

```golang
type Extension interface {
  Component
}
```

Reply:
The reason it is, is the same reason why `GetExtension` change in this PR, is a breaking change.

Reply:
We do the alias now, and try to figure out how to "unalias" without breaking, may be impossible, but should be minimal impact if any.